src/options.rs

```diff
@@ -524,6 +524,16 @@ impl Options {
         self
     }
 
+    /// Set a custom `JetStream` API prefix from a domain.
+    ///
+    pub fn jetstream_api_prefix_from_domain(mut self, domain: &str) -> Self {
```

Comment:
fn name seems really long, better way to do this?

For Go client its options to getting the JetStream context which I like.

Reply:
I like it, would just say domain though like we do in Go client. Do APIPrefix() option and Domain() option. In Rust version of course.

Reply:
As a first pass, let's just embed `JetStreamOptions` into `Options` (this PR).